python/ray/serve/_private/autoscaling_state.py

```diff
@@ -265,18 +270,203 @@ def get_decision_num_replicas(
 
         return self.apply_bounds(decision_num_replicas)
 
-    def get_total_num_requests(self) -> float:
-        """Get average total number of requests aggregated over the past
-        `look_back_period_s` number of seconds.
+    def _collect_replica_running_requests(self) -> List[Dict[str, List]]:
+        """Collect running requests metrics from replicas for aggregation."""
+        metrics_timeseries_dicts = []
+
+        for replica_id in self._running_replicas:
+            if replica_id in self._replica_requests:
+                metrics_timeseries_dicts.append(
+                    {
+                        metric_name: self._replica_requests[replica_id].metrics[
+                            metric_name
+                        ]
+                        for metric_name in self._replica_requests[replica_id].metrics
+                        if metric_name
+                        == RUNNING_REQUESTS_KEY  # only collect running requests
```

Comment:
Wait, why even iterate here if you're only looking for one key? Wouldn't something like `{RUNNING_REQUESTS_KEY: self._replica_requests[replica_id].metrics[RUNNING_REQUESTS_KEY]}` do what you want?